src/main/java/com/cburch/logisim/fpga/data/IOComponentTypes.java

```diff
@@ -200,15 +244,41 @@ public static void getPartialMapInfo(Integer[][] PartialMap,
       int nrOfPins,
       int nrOfRows,
       int nrOfColumns,
+      int mapRotation,
       IOComponentTypes type) {
     var hasDp = false;
     switch (type) {
       case DIPSwitch: {
-        var part = (width > height) ? (float) width / (float) nrOfPins : (float) height / (float) nrOfPins;
-        for (var w = 0; w < width; w++)
-          for (var h = 0; h < height; h++) {
-            var index = (width > height) ? (float) w / part : (float) h / part;
-            PartialMap[w][h] = (int) index;
+        var part = (float) 0;
+        switch (mapRotation) {
+          case rotationPlusNinety :
+          case rotationMinusNinety : {
+            part = (float) height / (float) nrOfPins;
+            break;
+          }
+          default : {
+            part = (float) width / (float) nrOfPins;
+            break;
+          }
+        }
+        for (var widthIndex = 0; widthIndex < width; widthIndex++)
+          for (var heightIndex = 0; heightIndex < height; heightIndex++) {
+            var pinIndex = (float) 0;
```

Comment:
Maybe `var pinIndex = 0f;` or regular `float pinIndex = 0;` instead? Multiple places below.

But by logic, `pinIndex` should be `int` really, so fixes (rounding) should be in all math below instead of casting while adding to `PartialMap`